opensearch-operator/controllers/dashboard_test.go

```diff
@@ -119,5 +119,30 @@ var _ = Describe("Dashboards Reconciler", func() {
 			Expect(HasOwnerReference(&cm, &OpensearchCluster)).To(BeTrue())
 			Expect(HasOwnerReference(&service, &OpensearchCluster)).To(BeTrue())
 		})
+		It("should create and configure dashboard deployment correctly", func() {
+			if OpensearchCluster.Spec.Dashboards.Enable {
+				dashboardDeployName := fmt.Sprintf("%s-dashboards", OpensearchCluster.Name)
+				deployment := &sts.Deployment{}
+				Eventually(func() error {
+					return k8sClient.Get(context.Background(), types.NamespacedName{
+						Name:      dashboardDeployName,
+						Namespace: OpensearchCluster.Namespace,
+					}, deployment)
+				}, timeout, interval).Should(Succeed())
+				Expect(deployment.Spec.Replicas).To(Equal(2))
```

Comment:
You are comparing a pointer with a value, and the configured replicas is `3`. You can use the `k8s.io/utils/pointer` package and create a pointer:
```suggestion
				Expect(deployment.Spec.Replicas).To(Equal(pointer.Int32(3)))
```